rust-nightly-version

```diff
@@ -1 +1 @@
-nightly-2022-01-27
+nightly-2022-04-13
```

Comment:
how do we choose which nightly version to use?

Reply:
Good question! I chose the latest version at that time when I tried updating Rust toolchain.

Reply:
should we use a more recent one when this PR lands? eg 2022-05-28. I will check it has components for aarch64-apple-darwin as well

Reply:
I'm updating this to 1.61.0. I also tried later version of nightly, but ran into this https://github.com/rust-lang/rust-clippy/issues/8867 with some PoS types with generics. `2022-05-20` seems to be the last ok one